apps/asap-server/src/entities/research-output.ts

```diff
@@ -40,6 +40,24 @@ export const parseGraphQLResearchOutput = (
       }
     : {};
 
+
+  function userIsPMInTeam(user: GraphqlUser, team: GraphqlTeam): boolean {
+    const teamsMatchAndUserIsPM = (innerTeam: UserTeamConnection<GraphqlTeam>) =>
+      innerTeam.id[0].id === team.id && innerTeam.role === 'Project Manager';
+
+    const filteredTeams = user.flatData?.teams?.filter(teamsMatchAndUserIsPM);
```

Comment:
so... there is a small problem with your types - they are not aligned with the graphql query. Well not entirely at least, let me explain.

In here 
https://github.com/yldio/asap-hub/blob/6328a1b8cdd7753da8d2fdf652f12919a847aeb2/apps/asap-server/src/controllers/research-outputs.ts#L60-L64

you specify that you only want an `email` from a `User` but later on you have it declared as 

https://github.com/yldio/asap-hub/blob/f18503f576e1abeacb0a0a702b083309a5e96a0f/packages/squidex/src/entities/team.ts#L24

which suggests that we should get a full `GraphqlUser` from the relationship where in fact we only ever get the `email`. You probably did it cause you followed this:

https://github.com/yldio/asap-hub/blob/68ba60814eb35ea05cfc9b15e9d928a87f411919/packages/squidex/src/entities/research-output.ts#L49

Which makes sense (I'm pretty sure I told you to follow similar implementations) but unfortunately in your situation it will not work. The type logically is correct because every property of a `GraphqlUser` is marked as optional - except we know for a fact only `email` will ever be present so your filters will not work (`.teams` will always come as `undefined`).

Also when you say that your `referencingUsersContents` is of type `GraphqlUser` you are actually making it more difficult to filter users by a role as you are then having to traverse back through to the team the user belongs to. Instead you could create a new type for the user connection where alongside user email a `role` will be added. Check out this query:
```
{
  queryResearchOutputsContents {
    id
    referencingTeamsContents {
      referencingUsersContents {
        flatData {
          email
          role
        }
      }
    }
  }
}
```
in here our `referencingUsersContents` should point to a type which only has two properties `email` and `role` and you can use the role to limit it to just the PMs.

Reply:
I initially tried
```
flatData {
  email
  role
}
```
but it seems to come back with the wrong kind (?) of role: either `Guest` or `Grantee` and looking in `apps/asap-cli/src/import/users/parse.ts`, I guess possibly `Staff` also. The kind of role needed is in `packages/model/src/team.ts`. where `"Project Manager"` is defined on `TeamRole`. That being said in the PR env `pmsEmails` is always coming back empty `[]` so there may be an issue with the query.

Reply:
i explained why:
![image](https://user-images.githubusercontent.com/11645956/122791223-f451a300-d2b0-11eb-9e4f-82d2467a9e55.png)

try adding 
```
teams {
    role
    id {
      id
    }
}
```
to the flatData (where you said you tried adding just `role`)

Reply:
This works

Reply:
Manual testing:

From `https://918.hub.asap.science/shared-research`:

The two PM emails for the first shared RO `pmsEmails: ["test-email1@asap.science", "harlan.kohll@yld.io"]` are returned and the non-PM member is not.